src/service/status/InternalDeviceStatusService.cpp

```diff
@@ -158,13 +166,16 @@ void InternalDeviceStatusService::requestDevicesStatus()
     {
         auto keys = m_deviceRepository->findAllDeviceKeys();
 
+        m_selfRequestedDevices = {};
```

Comment:
If we have methods for all other forms of access to this container, then you should also clear this via method call